src/sentry/search/django/backend.py

```diff
@@ -269,6 +269,19 @@ def query(self, projects, tags=None, environments=None, sort_by='date', limit=10
             ),
             'active_at_from': ScalarCondition('active_at', 'gt'),
             'active_at_to': ScalarCondition('active_at', 'lt'),
+            'external_issue_id': CallbackCondition(
+                lambda queryset, external_issue_id: (queryset.filter if external_issue_id else queryset.exclude)(
+                    id__in=GroupLink.objects.filter(
+                        linked_type=GroupLink.LinkedType.issue,
+                        linked_id=ExternalIssue.objects.filter(
+                            # as per the initial check there must be 1 organization_id and at least
+                            # 1 project
+                            organization_id__in=projects[0].organization_id,
+                            key=external_issue_id,
```

Comment:
My concern with this was how the user would figure out what needs to go in this field. For a few integrations the key is not displayed anywhere in the UI. We could generate links but the added filter wouldn't be usable in the text search/advanced filter.